jerry-libc/include/assert.h

```diff
@@ -25,6 +25,11 @@ extern "C"
 #endif /* __cplusplus */
 
 #ifndef NDEBUG
+
+#ifndef __GNUC__
+#define __builtin_expect(expression, expected_value) (expression)
+#endif /* !__GNUC__ */
```

Comment:
@t-harvey Good point, I think in this case the pragmatic solution is to just drop the `__builtin_expect()` use from `assert.h`. Since this is only affects the code in jerry-libc (rather than jerry-core) and most of the `assert()` calls seem to be in the `printf()` implementation (which is not really performance-critical code) the performance impact should be negligible. Builds with asserts enabled are expected to have a noticeable performance overhead anyway.